tensorflow_addons/losses/giou_loss_test.py

```diff
@@ -40,62 +39,35 @@ def test_iou(dtype):
-        boxes2 = tf.constant(
-            [[3.0, 4.0, 6.0, 8.0], [14.0, 14.0, 15.0, 15.0]], dtype=dtype
-        )
-        expected_result = tf.constant(
-            [1.07500000298023224, 1.9333333373069763], dtype=dtype
-        )
-        loss = giou_loss(boxes1, boxes2)
-        self.assertAllCloseAccordingToType(loss, expected_result)
-
-    def test_with_integer(self):
-        boxes1 = tf.constant([[4, 3, 7, 5], [5, 6, 10, 7]], dtype=tf.int32)
-        boxes2 = tf.constant([[3, 4, 6, 8], [14, 14, 15, 15]], dtype=tf.int32)
-        expected_result = tf.constant(
-            [1.07500000298023224, 1.9333333373069763], dtype=tf.float32
-        )
-        loss = giou_loss(boxes1, boxes2)
-        self.assertAllCloseAccordingToType(loss, expected_result)
-
-    @parameterized.named_parameters(
-        ("float16", np.float16), ("float32", np.float32), ("float64", np.float64)
-    )
-    def test_different_shapes(self, dtype):
-        boxes1 = tf.constant([[4.0, 3.0, 7.0, 5.0], [5.0, 6.0, 10.0, 7.0]], dtype=dtype)
-        boxes2 = tf.constant([[3.0, 4.0, 6.0, 8.0]], dtype=dtype)
-        tf.expand_dims(boxes1, -2)
-        tf.expand_dims(boxes2, 0)
-        expected_result = tf.constant([1.07500000298023224, 1.366071], dtype=dtype)
-        loss = giou_loss(boxes1, boxes2)
-        self.assertAllCloseAccordingToType(loss, expected_result)
-
-        boxes1 = tf.constant([[4.0, 3.0, 7.0, 5.0], [5.0, 6.0, 10.0, 7.0]], dtype=dtype)
-        boxes2 = tf.constant([[3.0, 4.0, 6.0, 8.0]], dtype=dtype)
-        expand_boxes1 = tf.expand_dims(boxes1, -2)
-        expand_boxes2 = tf.expand_dims(boxes2, 0)
-        expected_result = tf.constant([1.07500000298023224, 1.366071], dtype=dtype)
-        loss = giou_loss(expand_boxes1, expand_boxes2)
-        self.assertAllCloseAccordingToType(loss, expected_result)
-
-    @parameterized.named_parameters(
-        ("float16", np.float16), ("float32", np.float32), ("float64", np.float64)
+@pytest.mark.parametrize("dtype", [np.float16, np.float32, np.float64])
+def test_giou_loss(dtype):
+    boxes1 = tf.constant([[4.0, 3.0, 7.0, 5.0], [5.0, 6.0, 10.0, 7.0]], dtype=dtype)
+    boxes2 = tf.constant([[3.0, 4.0, 6.0, 8.0], [14.0, 14.0, 15.0, 15.0]], dtype=dtype)
+    expected_result = tf.constant(
+        [1.07500000298023224, 1.9333333373069763], dtype=dtype
     )
-    def test_one_bbox(self, dtype):
-        boxes1 = tf.constant([4.0, 3.0, 7.0, 5.0], dtype=dtype)
-        boxes2 = tf.constant([3.0, 4.0, 6.0, 8.0], dtype=dtype)
-        expected_result = tf.constant(1.07500000298023224, dtype=dtype)
-        loss = giou_loss(boxes1, boxes2)
-        self.assertAllCloseAccordingToType(loss, expected_result)
+    loss = giou_loss(boxes1, boxes2)
+    test_utils.assert_allclose_according_to_type(loss, expected_result)
+
+
+@pytest.mark.parametrize("dtype", [np.float16, np.float32, np.float64])
+def test_different_shapes(dtype):
```

Comment:
It seems a part of the test is missing. Could you run the tests also with the boxes without expand_dims?

Reply:
Thanks @gabrieldemarmiesse, but without having expand_dims should already in `test_giou_loss`?

Reply:
You're right. My bad.